crates/swc_ecma_utils/src/node_ignore_span.rs

```diff
@@ -0,0 +1,134 @@
+use std::{borrow::Cow, fmt::Debug, hash::Hash};
+
+use swc_common::EqIgnoreSpan;
+use swc_ecma_ast::{Expr, Ident, MemberProp};
+
+/// A newtype that will ignore Span while doing `eq` or `hash`.
+pub struct NodeIgnoringSpan<'a, Node: ToOwned + Debug>(Cow<'a, Node>);
+
+impl<'a, Node: ToOwned + Debug> Debug for NodeIgnoringSpan<'a, Node> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("NodeIgnoringSpan").field(&*self.0).finish()
+    }
+}
+
+impl<'a, Node: ToOwned + Debug> NodeIgnoringSpan<'a, Node> {
+    #[inline]
+    pub fn borrowed(expr: &'a Node) -> Self {
+        Self(Cow::Borrowed(expr))
+    }
+
+    #[inline]
+    pub fn owned(expr: <Node as ToOwned>::Owned) -> Self {
+        Self(Cow::Owned(expr))
+    }
+}
+
+impl<'a, Node: EqIgnoreSpan + ToOwned + Debug> PartialEq for NodeIgnoringSpan<'a, Node> {
+    fn eq(&self, other: &Self) -> bool {
+        Ident::within_ignored_ctxt(|| self.0.eq_ignore_span(&other.0))
```

Comment:
I think we can move `Ident::within_ignored_ctxt` to the caller side.

Reply:
Sometimes we need to match on syntax context

Reply:
And actually, I think it's better to touch TLS less time. It's quite slow if you don't use LTO

Reply:
Done